embulk-api/src/main/java/org/embulk/spi/json/JsonString.java

```diff
@@ -84,6 +84,17 @@ public JsonString asJsonString() {
         return this;
     }
 
+    /**
+     * Returns the approximate size of this JSON string in bytes presumed to occupy in {@link org.embulk.spi.Page} as a reference.
+     *
+     * @return the approximate size of this JSON string in bytes presumed to occupy in {@link org.embulk.spi.Page} as a reference
+     */
+    @Override
+    public int presumeReferenceSizeInBytes() {
+        // Indeed, null is stored in Page as a special form as |nullBitSet|, but considered as 1 here in JsonValue just for ease.
+        return this.value.length() * 2 + 4;
```

Comment:
Minor question why return twice the length?

Reply:
It comes from :
> https://github.com/embulk/embulk/blob/v0.10.41/embulk-core/src/main/java/org/embulk/spi/PageBuilderImpl.java#L173

It's a difficult question in general to tell "how many bytes are consumed by a string". It depends on JVM's internal representation, and that could be different per a lot of perspective, including JVM versions. But it would be some reasonable persumption because JVM's internal representation of strings has been UTF-16 at least so far. (It may change, though.)

Reply:
Ah, indeed, Strings are not always UTF-16 already since Java 9. ;)

https://openjdk.org/jeps/254